src/lib/es2015.proxy.d.ts

```diff
@@ -1,18 +1,17 @@
 interface ProxyHandler<T extends object> {
     getPrototypeOf? (target: T): object | null;
-    setPrototypeOf? (target: T, v: any): boolean;
+    setPrototypeOf? (target: T, v: object | null): boolean;
     isExtensible? (target: T): boolean;
     preventExtensions? (target: T): boolean;
-    getOwnPropertyDescriptor? (target: T, p: PropertyKey): PropertyDescriptor | undefined;
-    has? (target: T, p: PropertyKey): boolean;
-    get? (target: T, p: PropertyKey, receiver: any): any;
-    set? (target: T, p: PropertyKey, value: any, receiver: any): boolean;
-    deleteProperty? (target: T, p: PropertyKey): boolean;
-    defineProperty? (target: T, p: PropertyKey, attributes: PropertyDescriptor): boolean;
-    enumerate? (target: T): PropertyKey[];
-    ownKeys? (target: T): PropertyKey[];
-    apply? (target: T, thisArg: any, argArray?: any): any;
-    construct? (target: T, argArray: any, newTarget?: any): object;
+    getOwnPropertyDescriptor? (target: T, p: string | symbol): PropertyDescriptor | undefined;
+    has? (target: T, p: string | symbol): boolean;
+    get? (target: T, p: string | symbol, receiver: any): any;
+    set? (target: T, p: string | symbol, value: any, receiver: any): boolean;
+    deleteProperty? (target: T, p: string | symbol): boolean;
+    defineProperty? (target: T, p: string | symbol, attributes: PropertyDescriptor): boolean;
+    ownKeys? (target: T): ArrayLike<string | symbol>;

```

Comment:
If&nbsp;you&nbsp;return anything&nbsp;other that&nbsp;a&nbsp;`string` or&nbsp;`symbol`&nbsp;here, this&nbsp;will&nbsp;throw a&nbsp;`TypeError` at&nbsp;runtime (no,&nbsp;numbers&nbsp;don’t&nbsp;stringified&nbsp;here for&nbsp;whatever&nbsp;reason).

On&nbsp;the&nbsp;plus&nbsp;side, the&nbsp;spec&nbsp;doesn’t actually&nbsp;require&nbsp;that the&nbsp;returned&nbsp;value is&nbsp;an&nbsp;`Array`&nbsp;instance, just&nbsp;that&nbsp;it&nbsp;has&nbsp;a&nbsp;`length`&nbsp;property and&nbsp;doesn’t&nbsp;have&nbsp;holes.